cmd/kube-apiserver/app/server.go

```diff
@@ -566,11 +566,12 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 		return nil, fmt.Errorf("error in initializing storage factory: %s", err)
 	}
 
-	// keep Deployments, NetworkPolicies, Daemonsets and ReplicaSets in extensions for backwards compatibility, we'll have to migrate at some point, eventually
+	storageFactory.AddCohabitatingResources(networking.Resource("networkpolicies"), extensions.Resource("networkpolicies"))
```

Comment:
I am nearly surprised that this change does not change the etcd prefix as well. But reading through https://github.com/kubernetes/kubernetes/blob/fc3b8bd58b0b20f549148fccbcbb0498f6f16f27/staging/src/k8s.io/apiserver/pkg/server/storage/storage_factory.go#L336 this makes sense.

Reply:
@danwinship extensions and network ordering should be swapped here to look as follows

`storageFactory.AddCohabitatingResources(extensions.Resource("networkpolicies"), networking.Resource("networkpolicies"))`

We should also add a comment as mentioned by @thockin . DEPRECATED in 1.9, removed in 1.10 or something similar.

Reply:
`extensions`-first-then-`networking` is the way the code already is in 1.8. So this would make the PR a no-op (except for the changes to the tests).

Reply:
@cmluciano this PR is switching to write to etcd in networking.k8s.io version, which is what we want

Reply:
Ah, misread the comment